volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -539,3 +545,248 @@ def reference_count(self):
+
+        # Unix socket states reuse (a subset) of the inet_sock states contants
+        if self.sk.type == "STREAM":
+            state_idx = self.sk.__sk_common.skc_state
+            if 0 <= state_idx < len(TCP_STATES):
+                state = TCP_STATES[state_idx]
+            else:
+                state = "UNKNOWN"
+        else:
+            state = "UNCONNECTED"
+
+        return state
+
+    @property
+    def inode(self):
+        return self.sk.inode
+
+class inet_sock(objects.StructType):
+    @property
+    def family(self):
+        family_idx = self.sk.__sk_common.skc_family
+        if 0 <= family_idx < len(SOCK_FAMILY):
+            return SOCK_FAMILY[family_idx]
+        else:
+            return "UNKNOWN"
+
+    @property
+    def protocol(self):
+        # If INET6 family and a proto is defined, we use that specific IPv6 protocol.
+        # Otherwise, we use the standard IP protocol.
+        protocol = IP_PROTOCOLS.get(self.sk.sk_protocol, "UNKNOWN")
+        if self.family == "AF_INET6":
+            protocol = IPV6_PROTOCOLS.get(self.sk.sk_protocol, protocol)
+        return protocol
+
+    @property
+    def state(self):
+        """Return a string representing the sock state."""
+
+        if self.sk.type == "STREAM":
+            state_idx = self.sk.__sk_common.skc_state
+            if 0 <= state_idx < len(TCP_STATES):
+                state = TCP_STATES[state_idx]
+            else:
+                state = "UNKNOWN"
+        else:
+            state = "UNCONNECTED"
+
+        return state
+
+    @property
+    def src_port(self):
+        sport_le = getattr(self, "sport", getattr(self, "inet_sport", None))
+        if sport_le is not None:
+            return socket.htons(sport_le)
+
+    @property
+    def dst_port(self):
+        sk_common = self.sk.__sk_common
+        if hasattr(sk_common, "skc_portpair"):
+            dport_le = sk_common.skc_portpair & 0xffff
+        elif hasattr(self, "dport"):
+            dport_le = self.dport
+        elif hasattr(self, "inet_dport"):
+            dport_le = self.inet_dport
+        elif hasattr(sk_common, "skc_dport"):
+            dport_le = sk_common.skc_dport
+        else:
+            return
+
+        return socket.htons(dport_le)
+
+    @property
+    def src_addr(self):
+        sk_common = self.sk.__sk_common
+        family = sk_common.skc_family
+        if family == socket.AF_INET:
+            addr_size = 4
+            if hasattr(self, "rcv_saddr"):
+                saddr = self.rcv_saddr
+            elif hasattr(self, "inet_rcv_saddr"):
+                saddr = self.inet_rcv_saddr
+            else:
+                saddr = sk_common.skc_rcv_saddr
+        elif family == socket.AF_INET6:
+            addr_size = 16
+            saddr = self.pinet6.saddr
+        else:
+            return
+
+        parent_layer = self._context.layers[self.vol.layer_name]
+        addr_bytes = parent_layer.read(saddr.vol.offset, addr_size)
```

Comment:
This needs a try/except for invalid addresses.

Reply:
hm.. what should it be the expected action if an invalid address exception is catched? return None and the plugins handle it? Shouldn't the plugin take care of not calling it with an invalid address?

Reply:
I am questioning the best way to handle this as its a bit more complicated than most other @property functions. The issue is that if a property function throws an exception, the the calling plugin will get a really strange error that the method does not exist (src_addr) in this case, when obviously it does. 

@ikelos given that this property function does more work than just retrieving a member, would this be better as a method like get_src_addr() that the plugins would call? Then it would be more normal to try/except the function call from the plugins.

Reply:
Yeah, if there's a chance it may give a bogus result then it'll either need to return None (potentially a bit confusing and would require as many tests as wrapping the call in a try/except), or I think better would be @atcuno 's suggestion to go with `get_src_addr`.

Reply:
I understand your point. if we convert the **src_addr** property to a method **get_src_addr** that will also require to change **dst_addr** and, for consistency, **src_port** and **dst_port**. And probably the others properties **family**, **protocol** and **state**. Is that all ok?

Reply:
yes, I think any of them that have logic beyond determining which member exists in the symbol table should be converted.

Reply:
done, properties changed to just getters